api/services/post.js

```diff
@@ -134,6 +136,65 @@ function _getServiceConfig(bag, next) {
   );
 }
 
+function _getMaster(bag, next) {
+  var who = bag.who + '|' + _getMaster.name;
+  logger.verbose(who, 'Inside');
+
+  configHandler.get('master',
+    function (err, master) {
+      if (err)
+        return next(
+          new ActErr(who, ActErr.DataNotFound,
+            'Failed to get master', err)
+        );
+
+      if (_.isEmpty(master))
+        return next(
+          new ActErr(who, ActErr.DataNotFound,
+            'No configuration in database for master')
+        );
+
+      bag.master = master;
+      return next();
+    }
+  );
+}
+
+function _getWorkers(bag, next) {
+  var who = bag.who + '|' + _getWorkers.name;
+  logger.verbose(who, 'Inside');
+
+  configHandler.get('workers',
+    function (err, workers) {
+      if (err)
+        return next(
+          new ActErr(who, ActErr.DataNotFound,
+            'Failed to get ' + bag.component, err)
+        );
+
+      if (_.isEmpty(workers))
+        return next(
+          new ActErr(who, ActErr.DataNotFound,
+            'No configuration in database for ' + bag.component)
+        );
+
+      bag.isSwarmClusterInitialized = false;
+
+      // if even one worker has external IP then swarm cluster is initialized
+      _.each(bag.workers,
```

Comment:
can use a _.some here, if you only need to find one worker whose IP doesn't match the master